wallet/src/main/java/com/android/identity_credential/wallet/presentation/PresentationFlowActivity.kt

```diff
@@ -0,0 +1,294 @@
+            )
+
+            val issuerAuthCoseSign1 = Cbor.decode(staticAuthData.issuerAuth).asCoseSign1
+            val encodedMsoBytes = Cbor.decode(issuerAuthCoseSign1.payload!!)
+            val encodedMso = Cbor.encode(encodedMsoBytes.asTaggedEncodedCbor)
+            val mso = MobileSecurityObjectParser(encodedMso).parse()
+
+            val documentGenerator = DocumentGenerator(
+                mso.docType,
+                staticAuthData.issuerAuth,
+                presentationRequestData.sessionTranscript
+            )
+            documentGenerator.setIssuerNamespaces(mergedIssuerNamespaces)
+            return documentGenerator
+        }
+
+        /**
+         * Try adding adding the [Document], generated from [DocumentGenerator], to the device
+         * response and retyrn the resulting bytes. If the passed credential has a locked auth key
+         * then a [KeyLockedException] will be thrown
+         *
+         *
+         * @param documentGenerator instance of [DocumentGenerator] that generates the [Document]
+         * @param credential the [MdocCredential] to use for signing the generated [Document]
+         * @return the bytes of the response document to be sent to the device.
+         * @throws [KeyLockedException] if the credential has a locked auth key.
+         */
+        suspend fun addDocumentToDeviceResponse(
+            documentGenerator: DocumentGenerator,
+            credential: MdocCredential,
+            keyUnlockData: KeyUnlockData?,
+        ): ByteArray = suspendCancellableCoroutine { continuation ->
+            val deviceResponseGenerator =
+                DeviceResponseGenerator(Constants.DEVICE_RESPONSE_STATUS_OK)
+            try {
+                documentGenerator.updateDeviceNamespacesSignature(credential, keyUnlockData)
+                deviceResponseGenerator.addDocument(documentGenerator.generate())
+                continuation.resume(deviceResponseGenerator.generate())
+            } catch (e: Exception) {
+                continuation.resumeWithException(e)
+            }
+        }
+
+        /**
+         * Helper function extending [DocumentGenerator] that updates the signature used to sign the
+         * generated [Document]. Throws a [KeyLockedException] if the authentication key is locked.
+
+         * @param credential the [MdocCredential] to use (for a secure area).
+         * @param unlockData data used to create the signature for signing the key in a secure area
+         * @throws [KeyLockedException] if the authentication key is locked.
+         */
+        fun DocumentGenerator.updateDeviceNamespacesSignature(
+            credential: MdocCredential,
+            unlockData: KeyUnlockData?
+        ) {
+            setDeviceNamespacesSignature(
+                NameSpacedData.Builder().build(),
+                credential.secureArea,
+                credential.alias,
+                unlockData,
+                Algorithm.ES256
+            )
+        }
```

Comment:
Again, why are you adding an extension for something used exactly _once_? Please just move that single call to `setDeviceNamespacesSignature()` into the callsite on L256.